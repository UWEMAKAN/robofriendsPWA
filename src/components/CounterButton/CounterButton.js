import React, { Component } from 'react';

import classes from './CounterButton.module.css';

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.count !== nextState.count;
  }

  render() {
    return (
      <button className={ `${classes.CounterButton} pa3 ba b--green bg-lightest-blue` }
        id='counter'
        color={ this.props.color }
        onClick={ () => this.setState(state => ({ count: state.count + 1 }))}>
        Count: { this.state.count }
      </button>
    );
  }
}

export default CounterButton;