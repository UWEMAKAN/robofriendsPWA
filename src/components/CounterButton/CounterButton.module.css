.CounterButton {
  outline: none;
  cursor: pointer;
  background-color: #CDECFF;
  padding: 20px 40px;
  font-size: 20px;
}