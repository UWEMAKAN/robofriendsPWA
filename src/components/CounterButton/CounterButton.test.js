import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('CounterButton Component', () => {
  it('expect to render CounterButton Component', () => {
    const mockColor = 'steelblue';
    expect(shallow(<CounterButton color={ mockColor } />)).toMatchSnapshot();
  });

  it('expect to correctly increment counter', () => {
    const mockColor = 'steelblue';
    const wrapper = shallow(<CounterButton color={ mockColor } />);
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 1 });
    expect(wrapper.props().color).toEqual('steelblue');

  });
});