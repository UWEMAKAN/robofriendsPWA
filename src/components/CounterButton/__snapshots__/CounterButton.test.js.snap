// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component expect to render CounterButton Component 1`] = `
<button
  className="CounterButton pa3 ba b--green bg-lightest-blue"
  color="steelblue"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
